src/controllers/TalksController.php

```diff
@@ -123,35 +133,179 @@ public function postAction($request, $db) {
+                    if ($new_id) {
+                        $comment    = $comment_mapper->getCommentById($new_id);
+                        $talk       = $talk_mapper->getTalkById($talk_id);
+                        $speakers   = $talk_mapper->getSpeakerEmailsByTalkId($talk_id);
                         $recipients = array();
-                        foreach($speakers as $person) {
+                        foreach ($speakers as $person) {
                             $recipients[] = $person['email'];
                         }
-                        $emailService = new TalkCommentEmailService($this->config, $recipients, $talk, $comment);
+                        $emailService = new TalkCommentEmailService($this->config,
+                            $recipients, $talk, $comment);
                         $emailService->sendEmail();
                         $uri = $request->base . '/' . $request->version . '/talk_comments/' . $new_id;
                         header("Location: " . $uri, true, 201);
                         exit;
                     } else {
-                        throw new Exception("The comment could not be stored", 400);
+                        throw new Exception("The comment could not be stored",
+                            400);
                     }
                 case 'starred':
                     // the body of this request is completely irrelevant
                     // The logged in user *is* attending the talk.  Use DELETE to unattend
                     $talk_mapper = new TalkMapper($db, $request);
                     $talk_mapper->setUserStarred($talk_id, $request->user_id);
-                    header("Location: " . $request->base . $request->path_info, NULL, 201);
+                    header("Location: " . $request->base . $request->path_info,
+                        null, 201);
                     exit;
                 default:
                     throw new Exception("Operation not supported, sorry", 404);
             }
         } else {
-            throw new Exception("method not yet supported - sorry");
+            throw new Exception('This should never have happened, sorry', 500);
+        }
+    }
+
+    /**
+     * Create a new Talk.
+     *
+     * This method creates a new talk after being called via the URL
+     * "/events/<ID>/talks" or "/talks" with the event-id being part of the
+     * POST-values
+     *
+     * @param Request $request
+     * @param DB $db
+     *
+     * @throws Exception on different occasions
+     * @return array|bool
+     */
+    public function createTalkAction($request, $db)
+    {
+        // Set the event-ID either via URL or via POST-Data depending on how the
+        // method was called.
+        // When it's called via a POST to /talks/ the event-id is part of the
+        // POST-data, when it's called via POST to /events/<id>/talks/ the
+        // event-id is part of the URL but will be overwritten by POST-data
+        $event_id = null;
```

Comment:
The documentation says:

> To create a talk, you actually need to make a POST request to the *talks* collection of the *event* that the session belongs to

i.e. POSTing to /talks/ doesn't create a new talk. Do we want it to?